packages/hoprd/src/api/v2.ts

```diff
@@ -120,41 +113,65 @@ export function setupRestApi(
   }) as express.ErrorRequestHandler)
 }
 
-export function setupWsApi(server: WebSocketServer, node: Hopr, logs: LogStream, options: { apiToken?: string }) {
-  server.on('connection', (socket, req) => {
-    if (!authenticateWsConnection(logs, req, options.apiToken)) {
-      socket.send(
-        JSON.stringify({
-          type: 'auth-failed',
-          msg: 'authentication failed',
-          ts: new Date().toISOString()
-        })
-      )
-      socket.close()
+const WS_PATHS = {
+  NONE: '', // used for testing
+  MESSAGES: '/api/v2/messages/websocket'
+}
+
+export function setupWsApi(server: Server, wss: WebSocketServer, node: Hopr, options: { apiToken?: string }) {
+  // before upgrade to WS, we perform various checks
+  server.on('upgrade', function upgrade(req, socket, head) {
+    debugLog('WS client attempt to upgrade')
+    const path = removeQueryParams(req.url)
+    const needsAuth = !!options.apiToken
+
+    // check if path is supported
+    if (!Object.values(WS_PATHS).includes(path)) {
+      debugLog(`WS client path '${path}' does not exist`)
+      socket.end('HTTP/1.1 404 Not Found\r\n\r\n', () => socket.destroy())
```

Comment:
How come we don't use socket close here?

Reply:
I used this to guarantee termination after reading [this](https://stackoverflow.com/a/7341738), but to be totally honest it might be an overkill. Open to suggestions.

Reply:
This covers the http lib, but not the ws lib. Also we don't deploy, at least currently, hoprd behind reverse proxies so that keep alive setup won't be happening. I'd rather stick to the normal path until we experience bugs which warrant custom solutions.

Reply:
At this stage, we haven't upgraded yet so this socket is not a WS one. Close does not exist, I guess the analogous one would be `.end`.